### PR TITLE
Play state

### PR DESCRIPTION
**Pull request by Abhishek-90**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@Abhishek-90 

## Build fail log


[12:25:09.308] Cloning github.com/atapas/react-play (Branch: playState, Commit: ebbe0da)
[12:25:09.906] Cloning completed: 597.755ms
[12:25:10.331] Installing build runtime...
[12:25:11.888] Build runtime installed: 1.557s
[12:25:12.570] Looking up build cache...
[12:25:15.753] Build cache downloaded [40.35 MB]: 2742.012ms
[12:25:15.884] Detected package.json
[12:25:15.885] Installing dependencies...
[12:25:16.132] yarn install v1.22.17
[12:25:16.214] [1/4] Resolving packages...
[12:25:16.684] [2/4] Fetching packages...
[12:25:33.280] [3/4] Linking dependencies...
[12:25:33.283] warning " > @testing-library/user-event@13.5.0" has unmet peer dependency "@testing-library/dom@>=7.21.4".
[12:25:33.284] warning "react-helmet > react-side-effect@2.1.1" has incorrect peer dependency "react@^16.3.0 || ^17.0.0".
[12:25:33.285] warning "react-organizational-chart > @emotion/css > @emotion/babel-plugin@11.7.2" has unmet peer dependency "@babel/core@^7.0.0".
[12:25:33.285] warning "react-organizational-chart > @emotion/css > @emotion/babel-plugin > @babel/plugin-syntax-jsx@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
[12:25:33.288] warning "react-scripts > tailwindcss@3.0.12" has unmet peer dependency "autoprefixer@^10.0.2".
[12:25:33.291] warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-syntax-flow@^7.14.5".
[12:25:33.291] warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-transform-react-jsx@^7.14.9".
[12:25:33.295] warning "react-scripts > react-dev-utils > fork-ts-checker-webpack-plugin@6.5.0" has unmet peer dependency "typescript@>= 2.7".
[12:25:33.303] warning "react-scripts > eslint-config-react-app > @typescript-eslint/eslint-plugin > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
[12:25:33.305] warning " > react-shimmer-effect@1.0.9" has incorrect peer dependency "react@^16.8.1".
[12:25:33.305] warning " > react-shimmer-effect@1.0.9" has incorrect peer dependency "react-dom@^16.8.1".
[12:25:49.219] [4/4] Building fresh packages...
[12:25:56.485] Done in 40.36s.
[12:25:56.705] Running "yarn run build"
[12:25:56.885] yarn run v1.22.17
[12:25:56.912] $ react-scripts build
[12:25:58.163] Creating an optimized production build...
[12:26:24.606] Failed to compile.
[12:26:24.606] 
[12:26:24.606] src/meta/play-meta.js
[12:26:24.606] Syntax error: Identifier 'States' has already been declared. (13:0) (13:undefined)
[12:26:24.606] 
[12:26:24.606] 
[12:26:24.666] error Command failed with exit code 1.
[12:26:24.666] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[12:26:24.673] Error: Command "yarn run build" exited with 1

File: src/meta/play-meta.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ import {
   RandomMemeGenerator,
   ReactTodoApp,
   //import play here
+States,
```
Comment (atapas) [inline src/meta/play-meta.js:13]: Why have you added states again?

File: src/meta/play-meta.js [hunk 2]
```diff
@@ -80,10 +82,11 @@ export const plays = [
     github: 'murtuzaalisurti',
     featured: true
   }, {
-    id: 'pl-states',
-    name: 'States',
+    id: 'pl-states',
```
Comment (atapas) [inline src/meta/play-meta.js:85]: This file shouldn't have any changes. Please revert with main.

File: src/plays/states/States.jsx [hunk 3]
```diff
@@ -3,10 +3,9 @@ import { getPlayById } from "meta/play-meta-util";
 import PlayHeader from "common/playlists/PlayHeader";
 import "./states.css";
 
-function States(props) {
-  // Do not remove the below lines.
-  // The following code is to fetch the current play from the URL
-  const { id } = props;
+function States() {
+  const location = useLocation();
```
Comment (atapas) [inline src/plays/states/States.jsx:7]: We are not using useLocation any more. The code was refactored to fix the deeplinking bug. This changes reverts the bug fixing.

File: src/plays/states/States.jsx [hunk 4]
```diff
@@ -34,6 +33,21 @@ function States(props) {
   const handleDurationChange = (e) => {
     setDuration(e.target.value);
   };
+=======
```
Comment (atapas) [inline src/plays/states/States.jsx:36]: Merge conflict is not resolved correctly.

**Review by atapas — CHANGES_REQUESTED** [review]
@Abhishek-90 You haven't resolved merge conflicts correctly.. hence lots of unnecessary changes went in. Please take care. If you have issues in resolving merge conflicts, see some videos on how to do them. If the problem persists, scrap this branch and PR and create a new fresh one.

**Comment by Abhishek-90** [comment]
@atapas I have push the new changes with conflicts resolved

File: src/meta/play-meta.js [hunk 5]
```diff
@@ -130,4 +130,3 @@ export const plays = [
 ];
```
Comment (atapas) [inline src/meta/play-meta.js:130]: This file must not be committed. As there are no changes expected.

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by atapas** [comment]
> @atapas I have push the new changes with conflicts resolved

I see my older comments are not taken care.

File: src/meta/play-meta.js (hunk 5 above)
Comment (Abhishek-90) [reply src/meta/play-meta.js:130]: As I had commit with conflicts, I copied the contents from your main branch to pasted it here.

**Comment by Abhishek-90** [comment]
@atapas can you please elaborate ?

File: src/meta/play-meta.js (hunk 5 above)
Comment (atapas) [reply src/meta/play-meta.js:130]: No, that's not how you should resolve conflict. It should be by the portion of change between two commits, not by replacing the entire file content.

**Comment by atapas** [comment]
> @atapas can you please elaborate ?

`state.jsx` file.

File: src/plays/states/States.jsx (hunk 3 above)
Comment (Abhishek-90) [reply src/plays/states/States.jsx]: Okay, I understood what mistake I did. I think I have resolved the conflicts.

File: src/meta/play-meta.js [hunk 6]
```diff
@@ -127,7 +126,6 @@ export const plays = [
     blog: '',
     video: ''
   }, //replace new play item here
-];
-
+]
```
Comment (atapas) [inline src/meta/play-meta.js:129]: Why this semicolon was removed?

**Review by atapas — CHANGES_REQUESTED** [review]
Given a comment

+

You are comitting yarn lock and package lock, why? Please do not commit these files if you haven't changed the package.json file. If you pull the latest from the main, these two files shouldn't appear in your to be committed list.

File: src/meta/play-meta.js (hunk 6 above)
Comment (Abhishek-90) [reply src/meta/play-meta.js]: Added Semi Colon

**Review by atapas — COMMENTED** [review]
Ok, I'm noticing something. Let me know about it..

I see two CSS file one with capital case and another one with small case? Why do we need two files of same name with different cases? Why cant it be just one state.css file?

Also I still see yarn.lock in the committed list

image

**Comment by Abhishek-90** [comment]
@atapas Initially I used State.css but changed it to state.css on your suggestion. I meant to delete it from Github, Now I have deleted State.css

about yarn.lock, I deleted it from committed list, I am not sure why it is still showing

**Review by atapas — APPROVED** [review]

**Pull request merged**